<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport"
        content="width=device-width,initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no"/>
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="stylesheet" href="../assert.css">
  <script src="../assert.js"></script>
  <title>Promise</title>
</head>
<body>



<section class="d">
  <bar id="quadrangle">四边形 <div>33</div></bar>
  <h3 id="circle">圆</h3>
  <div id="triangle">三角形</div>
</section>x

<script>
  // 异常隐匿拒绝一个Promise
  const promise = new Promise((resolve, reject) => {
    undeclaredVariable++
  })

  promise.then(() => {
    fail('Happy path, wo not be called.')
  }, err => {
    fail('exception:--' + err)
    fail(err)
  }).catch(error => {
    fail('---Third promise was also rejected.----' + error)
  })
</script>

</body>
</html>